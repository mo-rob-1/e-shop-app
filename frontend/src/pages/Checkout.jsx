import { useState } from "react";
import { useSelector } from "react-redux";
import StripeContainer from "../components/StripeContainer";

function Checkout() {
  const [disabled, setDisabled] = useState(false);

  const cart = useSelector((state) => state.cart);
  const { cartItems } = cart;

  const saveAddress = (e) => {
    e.preventDefault();
    setDisabled(!disabled);
  };

  return (
    <>
      <h1 className="md:mb-10 font-bold text-2xl">Checkout</h1>

      <div className="checkout-container">
        <div>
          <h2 className="font-bold text-lg">Your Order</h2>
          <div className="grid gap-6 mt-8">
            {cartItems.map((item) => (
              <div
                key={item.product}
                className="shadow rounded-bl-xl rounded-br-xl rounded-tr-xl product-item-container items-center"
              >
                <div className="md:h-full">
                  <img
                    src={item.imageUrl}
                    alt={item.name}
                    className="rounded-tl-xl rounded-tr-xl w-full md:h-40 md:rounded-tr-none md:rounded-bl-xl object-cover"
                  />
                </div>

                <div className="pl-4 pr-4 pt-6 pb-6">
                  <h3 className="font-bold mb-2">{item.name}</h3>
                  <p className="font-bold mb-2">Â£{item.price}</p>
                  <p>
                    <span className="font-bold">Quantity:</span> {item.qty}
                  </p>
                </div>
              </div>
            ))}
          </div>
        </div>

        <div>
          <form className="shadow rounded-2xl p-6 mt-8 mb-8">
            <h2 className="font-bold mb-6 mt-2 text-lg">Shipping Address</h2>
            <div className="mb-4">
              <label className="font-semibold">Name:</label>
              <br></br>
              <input type="text" name="name" disabled={disabled} className="border w-full p-2 mt-1" />
            </div>
            <div className="mb-4">
              <label className="font-semibold">Address:</label> <br></br>
              <input type="text" name="address" disabled={disabled} className="border w-full p-2 mt-1" />
            </div>
            <div className="mb-4">
              <label className="font-semibold">City:</label> <br></br>
              <input type="text" name="city" disabled={disabled} className="border w-full p-2 mt-1" />
            </div>
            <div className="mb-4">
              <label className="font-semibold">Post Code:</label> <br></br>
              <input type="text" name="post" disabled={disabled} className="border w-full p-2 mt-1" />
            </div>
            <div className="mb-5">
              <label className="font-semibold">Country:</label> <br></br>
              <input type="text" name="country" disabled={disabled} className="border w-full p-2 mt-1" />
            </div>
            {disabled ? (
              <button
                type="submit"
                onClick={saveAddress}
                className="bg-black text-white font-bold inline-block pl-10 pr-10 pt-4 pb-4 mt-2 text-sm rounded-md"
              >
                Edit
              </button>
            ) : (
              <button
                type="submit"
                onClick={saveAddress}
                className="bg-slate-800 text-white font-semibold inline-block pl-10 pr-10 pt-4 pb-4 mt-2 text-sm rounded-md"
              >
                Save
              </button>
            )}
          </form>

          <div className="shadow rounded-2xl p-6">
            <h2 className="font-bold mb-6 mt-2 text-lg">Make Payment</h2>
            <StripeContainer />
          </div>
        </div>
      </div>
    </>
  );
}

export default Checkout;
