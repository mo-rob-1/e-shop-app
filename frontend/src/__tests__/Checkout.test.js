import { render } from "@testing-library/react";
import Checkout from "../pages/Checkout";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import store from "../redux/store";
import { BrowserRouter as Router } from "react-router-dom";

describe("Checkout", () => {
  it("renders the checkout page", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders checkout title text", () => {
    const { getByText } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByText("Checkout")).toBeInTheDocument();
  });

  it("renders your order title text", () => {
    const { getByText } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByText("Your Order")).toBeInTheDocument();
  });

  it("renders shipping address title text", () => {
    const { getByText } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByText("Shipping Address")).toBeInTheDocument();
  });

  it("renders the shipping address form", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("shipping-address-form")).toBeInTheDocument();
  });

  it("renders the name form group", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("name-form-group")).toBeInTheDocument();
  });

  it("renders the name label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("name-label")).toBeInTheDocument();
  });

  it("renders the name input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("name-input")).toBeInTheDocument();
  });

  it("renders the address form group", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("address-form-group")).toBeInTheDocument();
  });

  it("renders the address label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("address-label")).toBeInTheDocument();
  });

  it("renders the address input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("address-input")).toBeInTheDocument();
  });

  it("renders the city form group", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("city-form-group")).toBeInTheDocument();
  });

  it("renders the city label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("city-label")).toBeInTheDocument();
  });

  it("renders the city input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("city-input")).toBeInTheDocument();
  });

  it("renders the post code form group", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("post-code-form-group")).toBeInTheDocument();
  });

  it("renders the post code label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("post-code-label")).toBeInTheDocument();
  });

  it("renders the post code input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("post-code-input")).toBeInTheDocument();
  });

  it("renders the country form group", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("country-form-group")).toBeInTheDocument();
  });

  it("renders the country label", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("country-label")).toBeInTheDocument();
  });

  it("renders the country input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("country-input")).toBeInTheDocument();
  });

  it("renders the form button", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("form-button")).toBeInTheDocument();
  });

  it("renders the make payment form", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Checkout />
        </Router>
      </Provider>
    );
    expect(getByTestId("make-payment-form")).toBeInTheDocument();
  });
});
