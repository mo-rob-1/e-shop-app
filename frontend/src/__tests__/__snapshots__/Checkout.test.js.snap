// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout renders the checkout page 1`] = `
<DocumentFragment>
  <h1
    class="md:mb-10 font-bold text-2xl"
  >
    Checkout
  </h1>
  <div
    class="checkout-container"
  >
    <div>
      <h2
        class="font-bold text-lg"
      >
        Your Order
      </h2>
      <div
        class="grid gap-6 mt-8"
      />
    </div>
    <div>
      <form
        class="shadow rounded-2xl p-6 mt-8 mb-8"
        data-testid="shipping-address-form"
      >
        <h2
          class="font-bold mb-6 mt-2 text-lg"
        >
          Shipping Address
        </h2>
        <div
          class="mb-4"
          data-testid="name-form-group"
        >
          <label
            class="font-semibold"
            data-testid="name-label"
          >
            Name:
          </label>
          <br />
          <input
            class="border w-full p-2 mt-1"
            data-testid="name-input"
            name="name"
            type="text"
          />
        </div>
        <div
          class="mb-4"
          data-testid="address-form-group"
        >
          <label
            class="font-semibold"
            data-testid="address-label"
          >
            Address:
          </label>
           
          <br />
          <input
            class="border w-full p-2 mt-1"
            data-testid="address-input"
            name="address"
            type="text"
          />
        </div>
        <div
          class="mb-4"
          data-testid="city-form-group"
        >
          <label
            class="font-semibold"
            data-testid="city-label"
          >
            City:
          </label>
           
          <br />
          <input
            class="border w-full p-2 mt-1"
            data-testid="city-input"
            name="city"
            type="text"
          />
        </div>
        <div
          class="mb-4"
          data-testid="post-code-form-group"
        >
          <label
            class="font-semibold"
            data-testid="post-code-label"
          >
            Post Code:
          </label>
           
          <br />
          <input
            class="border w-full p-2 mt-1"
            data-testid="post-code-input"
            name="post"
            type="text"
          />
        </div>
        <div
          class="mb-5"
          data-testid="country-form-group"
        >
          <label
            class="font-semibold"
            data-testid="country-label"
          >
            Country:
          </label>
           
          <br />
          <input
            class="border w-full p-2 mt-1"
            data-testid="country-input"
            name="country"
            type="text"
          />
        </div>
        <div
          data-testid="form-button"
        >
          <button
            class="bg-slate-800 text-white font-semibold inline-block pl-10 pr-10 pt-4 pb-4 mt-2 text-sm rounded-md"
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
      <div
        class="shadow rounded-2xl p-6"
        data-testid="make-payment-form"
      >
        <h2
          class="font-bold mb-6 mt-2 text-lg"
        >
          Make Payment
        </h2>
        <form>
          <fieldset>
            <div
              class="mb-5"
            >
              <div />
            </div>
          </fieldset>
          <button
            class="bg-slate-800 text-white font-semibold inline-block pl-10 pr-10 pt-4 pb-4 mt-2 text-sm rounded-md"
          >
            Pay
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
